src/coreclr/vm/peassembly.inl

```diff
@@ -275,7 +275,7 @@ inline IMDInternalImport* PEAssembly::GetMDImport()
 
 #ifndef DACCESS_COMPILE
 
-inline IMetaDataImport2 *PEAssembly::GetRWImporter()
```

Comment:
I think `bool openForWriting` may be a better name for the bool flag here and in other places. The argument name should describe what the caller gets, not how it is implemented internally.

Reply:
This is the part that's interesting. The caller always gets a RWImporter. The effects are to the assembly cached fields only, so I didn't quite use the open* naming. If you think that's clearer, I'm not against it - it's just one of those cases where it feels the class has become leaky and it's purely needing to describe internal behavior. I'll add a more thorough comment to the function headers too.

Reply:
Yes, the caller always gets a RWImporter and we depend on the caller to not emit anything if the importer is not opened for writing. In theory, we enforce it by returning an error from the emit APIs until the importer is opened for writing. It would be an overkill and it would make the change much more complicated, but it would not change the contract.

Reply:
I think `bool openForWriting` name is cleaner.